lib/smart_answer_flows/benefit-cap-calculator.rb

```diff
@@ -121,26 +159,53 @@ def define
         end
 
         next_node do
-          question :single_couple_lone_parent?
+          if chosen_cap == "future"
+            question :single_couple_lone_parent_future?
+          else
+            question :single_couple_lone_parent?
+          end
         end
       end
 
-      #Q6
+      #Q6 current flow
       multiple_choice :single_couple_lone_parent? do
         precalculate :weekly_benefit_cap_descriptions do
-          config.weekly_benefit_cap_descriptions
+          config.weekly_benefit_cap_descriptions(chosen_cap)
+        end
```

Comment:
I think the `weekly_benefit_cap_descriptions` block is common to both versions of Q6. And since we always arrive at these via Q5p, we could move this block to be a `calculate` block in Q5p to avoid the duplication. As before, I would do this in a separate commit i.e. introduce duplication first and then remove it.